torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -55,21 +55,52 @@ Message deserialize(MessageType type, std::istream& is) {
 
 } // namespace
 
+void ProcessGroupAgent::collectNames() {
+  const std::string& workerName = workerId_.name_;
+  const auto worldSize = pg_->getSize();
+  const int64_t length = workerName.length();
+
+  // use c10d allgather to collect names
+  torch::Tensor nameTensor =
+      torch::zeros({WorkerId::MAX_NAME_LEN}, torch::kChar);
+  memcpy(nameTensor.storage().data(), workerName.c_str(), length);
```

Comment:
nit: can we just use `workerName.length()` here and get rid of the `length` variable?